addons/mass_mailing/models/mail_mail.py

```diff
@@ -51,7 +51,13 @@ def _get_unsubscribe_url(self, cr, uid, mail, email_to, context=None):
         url = urlparse.urljoin(
             base_url, 'mail/mailing/%(mailing_id)s/unsubscribe?%(params)s' % {
                 'mailing_id': mail.mailing_id.id,
-                'params': werkzeug.url_encode({'db': cr.dbname, 'res_id': mail.res_id, 'email': email_to})
+                'params': werkzeug.url_encode({
+                    'db': cr.dbname,
+                    'res_id': mail.res_id,
+                    'email': email_to,
+                    'token': self.pool['mail.mass_mailing']._unsubscribe_token(
+                        cr, uid, mail.mailing_id.id, mail.res_id, email_to),
```

Comment:
Then don't use `@api.multi`. If you use this decorator, you have to pass a list of IDs in old API. `self.ensure_one()` will check the single element in the list.